pkg/frontend/transport/handler.go

```diff
@@ -252,6 +252,52 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (f *Handler) parseURLValues(r *http.Request, stats *querier_stats.QueryStats, values url.Values) {
+	query := values.Get("query")
+	step := values.Get("step")
+	start := values.Get("start")
+	end := values.Get("end")
+	ts := values.Get("time")
+	var (
+		tsInt, startInt, endInt, stepInt int64
+		err                              error
+	)
+	stats.AddQuery(query)
+	if len(step) == 0 { // instant query
+		if len(ts) > 0 {
+			tsInt, err = util.ParseTime(ts)
+			if err != nil {
+				level.Error(util_log.WithContext(r.Context(), f.log)).Log("msg", "failed to parse time", "err", err)
+			}
+		} else {
+			tsInt = util.TimeToMillis(time.Now())
```

Comment:
Will `tsInt` be exactly the same as what querier sees ?

Reply:
good question. I think for instance query, if client side didn't pass time parameter, we will default to use the time.now()
as here : https://github.com/cortexproject/cortex/blob/master/pkg/querier/tripperware/instantquery/instant_query.go#L135

That timestamp is generated in the querier side when the query request is constructed. It will be little bit off here, because this time stamp is generated on the query-frontend side. But I don't see a problem, because it could be a few ms off, or at most a few seconds?